pkg/registry/secret/strategy.go

```diff
@@ -72,6 +72,23 @@ func (strategy) AllowUnconditionalUpdate() bool {
 	return true
 }
 
+func (s strategy) Export(obj runtime.Object, exact bool) error {
+	s.PrepareForCreate(obj)
+	if exact {
+		return nil
+	}
+	t, ok := obj.(*api.Secret)
+	if !ok {
+		// unexpected programmer error
+		return fmt.Errorf("unexpected object: %v", obj)
+	}
+	// secrets that are tied to the UID of a service account cannot be exported anyway
+	if t.Type == api.SecretTypeServiceAccountToken || len(t.Annotations[api.ServiceAccountUIDKey]) > 0 {
+		return fmt.Errorf("object omitted")
```

Comment:
Is this an error that a client should react to?  It might be more appropriate to return 422 invalid here, and indicate the field(s) that prevent the export.  Clients can then filter out everything unexportable.